core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1458,9 +1458,78 @@ class KafkaApis(val requestChannel: RequestChannel,
   }
 
   def handleTxnOffsetCommitRequest(request: RequestChannel.Request): Unit = {
```

Comment:
I spent about an hour looking into this. To factor out the common bits in a useful way, I think we will need to make `TxnOffsetCommitRequest` a subclass of `OffsetCommitRequest` (and also the corresponding response classes). 

Otherwise, the same information is represented differently in both, and trying to bridge the difference would be more code than is actually duplicated. I think we should leave the refactor as a future work item.